sherpa/astro/datastack/ds.py

```diff
@@ -108,47 +121,70 @@ def clear_stack(self):
             del _all_dataset_ids[dataid]
         self.__init__()
 
-    # QUS: should this use the logger instead of print?
-    def show_stack(self):
-        """Display basic information about the contents of the data stack.
+    def _summary_for_display(self):
+        '''Return information about the datastack used for printing
 
-        A brief summary of each data set in the stack is printed to the
-        standard output channel. The information displayed depends on the
-        type of each data set.
-        """
+        For each dataset in the datastack, id, name, and a number of
+        keys from the header (e.g. TELESCOP, MJD-OBS) are
+        collected. Then columns that are identical for all datasets
+        are removed (e.g. if no dataset has a GRATING keyword). That
+        way, the output can depend on the datasets loaded. In most
+        cases, only a few of a relatively long list of keywords will
+        differ between datasets, printing out only the relevant
+        information.
+
+        Returns
+        -------
+        header : list of strings
+            Column names
+        data : list of lists
+            Each list is a row, corresponding to one dataset
 
-        # The observation-date (MJD) format keyword was MJD_OBS but
-        # has changed (CIAO 4.12) for Chandra data to MJD-OBS, so
-        # need to look for both. The idea is to use the new key
-        # and then fall back to the old one.
-        #
+        '''
+        tab = []
         for dataset in self.filter_datasets():
-            obsid = "N/A"
-            time = "N/A"
-            timekey = 'MJD-OBS'
+            row = [dataset['id'], dataset['data'].name]
             try:
                 hdr = dataset['data'].header
+                row.extend([hdr.get(k, '') for k in keys_to_display])
+            except AttributeError:
+                row.append([''] * len(keys_to_display))
```

Comment:
I'm guessing this is to check for obects which don't have a header? Your recent changes mean that we don't getthis by default, but I guess a user could do something like `pha.header = None`.

Reply:
[This is not a request to change code, just so I can understand it as there's no test coverage]

Reply:
Yes.